packages/shelter-ui/src/tooltip.tsx

```diff
@@ -87,12 +94,53 @@ export function tooltip(el: HTMLElement, props: Accessor<JSX.Element | [boolean,
     }, 100);
   };
 
-  el.addEventListener("mouseenter", enterHandler);
-  el.addEventListener("mouseleave", exitHandler);
+  const [isInside, setIsInside] = createSignal(false);
+
+  const moveHandler = (clientX: number, clientY: number) => {
+    const bounding = el.getBoundingClientRect();
+    if (!bounding) {
+      setActive(false);
+      return;
+    }
+
+    // if inside the bounding box
+    if (
+      clientX > bounding.left &&
+      clientX < bounding.right &&
+      clientY > bounding.top &&
+      clientY < bounding.bottom
+    ) {
+      if (!isInside()) {
+        setIsInside(true);
+        enterHandler();
+      }
+    } else {
+      exitHandler();
+      setIsInside(false);
+    }
+  }
+
+  const mouseMoveHandler = (e: MouseEvent) => {
+    moveHandler(e.clientX, e.clientY);
+  }
+
+  const wheelHandler = (e: WheelEvent) => {
+    moveHandler(e.clientX, e.clientY);
+    exitHandler();
+    setTimeout(() => moveHandler(e.clientX, e.clientY), 100);
```

Comment:
this feels like its a hacky fix for something, is there a better way to go about this?

Reply:
actually settimeout pretty useless here I was testing something diffrent here.

Reply:
my guess looking at this is its a fix based on the code that handles animations (that's what the other 100ms timeout is for in this code) but not sure on that

Reply:
to be clear im not saying this is necessarily useless - just saying, did you test without it?

Reply:
I do not remember for sure. But it should work without it too. Becouse it actullay calls the exitHandler before doing that. It should not change anything visible.

Reply:
ah okay. well, i'll test myself anyway later on, can't rn

Reply:
Just tested now everything is working fine.